### PR TITLE
Format EditText output values with the supplied locale.  

### PR DESCRIPTION
**Pull request by rasheedsulayman**
This PR formats output values from the `CurrencyEditText` using the supplied `Locale`. The previous implementation wrongfully formats output values using an implicit locale that assumes that the decimal symbol will always be ".". This caused issues with locales/countries that don't have "." as the decimal symbol. 

Finally, the PRs adds support for getting user input as a `BigDecimal`

### CODE REVIEW

**Review by wadva474 — APPROVED** [review]

File: library/src/main/java/com/cottacush/android/currencyedittext/CurrencyEditText.kt [hunk 1]
```diff
@@ -76,12 +77,24 @@ class CurrencyEditText(context: Context, attrs: AttributeSet?) : TextInputEditTe
...
+        return parseMoneyValueWithLocale(
+            locale,
             text.toString(),
             textWatcher.decimalFormatSymbols.groupingSeparator.toString(),
             currencySymbolPrefix
-        ).toDoubleOrNull()
+        ).toDouble()
+    }
+
+    fun getNumericValueBigDecimal(): BigDecimal {
```
Comment (JubrilEdu) [inline library/src/main/java/com/cottacush/android/currencyedittext/CurrencyEditText.kt:89]: Awesome.

File: sample/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -27,8 +27,7 @@
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
```
Comment (JubrilEdu) [inline sample/src/main/res/layout/activity_main.xml:28]: you might want to extract the strings to the string resource file

File: sample/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -49,7 +48,7 @@
             android:layout_marginStart="8dp" app:layout_constraintEnd_toEndOf="parent" android:layout_marginEnd="8dp"
             app:layout_constraintVertical_bias="0.19999999"/>
     <Button
-            android:text="get value"
+            android:text="Get value"
```
Comment (JubrilEdu) [inline sample/src/main/res/layout/activity_main.xml]: Same as above

**Comment by rasheedsulayman** [comment]
The hardcoded String values have been extracted @djubreel

**Review by efguydan — APPROVED** [review]

**Comment by efguydan** [comment]
Nice work, @r4sh33d

**Pull request merged**